web/github.go

```diff
@@ -70,45 +132,159 @@ func (g *GitHub) Orgs(w http.ResponseWriter, r *http.Request) {
-			ctxlog.Get(r.Context()).Errorf(nil, "failed to get GitHub installation organization name")
-			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-			return
-		}
-
 		// a GitHub App can also be installed in a User account
-		if *installation.Account.Type != "Organization" {
+		if installation.GetAccount().GetType() != "Organization" {
 			continue
 		}
 
-		org := *installation.Account.Login
-		mem, _, err := client.Organizations.GetOrgMembership(r.Context(), login, org)
+		admin, err := g.admin(r.Context(), installation, login)
 		if err != nil {
-			ctxlog.Get(r.Context()).Errorf(err, "failed to get GitHub user %s membership to organization %s: ", login, org)
+			ctxlog.Get(r.Context()).Errorf(err, "failed to check user admin role")
 			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 			return
 		}
 
-		if *mem.Role == "admin" {
-			orgs = append(orgs, orgRespose{Name: org})
+		if admin {
+			orgs = append(orgs, orgsListItem{
+				ID:   installation.GetAccount().GetID(),
+				Name: installation.GetAccount().GetLogin(),
+			})
 		}
 	}
 
 	successJSON(w, r, orgs)
 }
+
+// TODO (@carlosms) check if this makes more sense as a middleware
+// orgInstallation returns the GitHub App Installation corresponding to the URL
+// parameter "orgName". The logged-in user must be an administrator of the
+// organization. If there is any error the proper HTTP headers are set in w.
+func (g *GitHub) orgInstallation(w http.ResponseWriter, r *http.Request) (*github.Installation, error) {
+	login, err := GetUserLogin(r.Context())
+	if err != nil {
+		ctxlog.Get(r.Context()).Errorf(err, "failed to get user login from context")
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return nil, fmt.Errorf(http.StatusText(http.StatusUnauthorized))
+	}
+
+	orgName := chi.URLParam(r, "orgName")
+	if orgName == "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return nil, fmt.Errorf(http.StatusText(http.StatusBadRequest))
+	}
+
+	installation, err := g.installation(r.Context(), orgName)
+	if err != nil {
+		if err == errNotFound {
+			http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+			return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+		}
+
+		ctxlog.Get(r.Context()).Errorf(err, "failed to get installation with name %v", orgName)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return nil, fmt.Errorf(http.StatusText(http.StatusInternalServerError))
+	}
+
+	// a GitHub App can also be installed in a User account
+	if installation.GetAccount().GetType() != "Organization" {
+		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+	}
+
+	admin, err := g.admin(r.Context(), installation, login)
```

Comment:
since `admin` returns a boolean, I wonder if it would be better `isAdmin`
```suggestion
	admin, err := g.isAdmin(r.Context(), installation, login)
```